Use native implementation for adjoints in (control) operations (#1063)
PennyLaneAI/pennylane-lightning: .github/CHANGELOG.md

### Improvements

> * Apply controlled-adjoint and adjoint-controlled operators with native C++ implementation

Reviewer: Have you got time to check the feature parity in Catalyst?

Author: I checked - for adjoint(gate), ctrl(adjoint(gate)) , adj(ctrl(gate)), Catalyst does call NamedOperation with the proper `inverse` flag in C++ (i.e. reduced to `NamedOperation(gate, inverse, controlled_wires)` correctly)

> * Apply controlled-adjoint and adjoint-controlled operators with native C++ implementation

Reviewer: ```suggestion
* Use native C++ kernels for controlled-adjoint and adjoint-controlled of supported operations.
```
